app/models/user.rb

```diff
@@ -34,6 +34,7 @@ class User < ActiveRecord::Base
   has_many :contestants, through: :contestants_users
 
   def name
-    "#{first_name} #{last_name}"
+    name = "#{first_name} #{last_name}"
+    name == ' ' ? self.email : name
```

Comment:
Redundant `self` detected.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.